test-packages/e2e-tests/test/on-prem/odata.spec.ts

```diff
@@ -0,0 +1,45 @@
+import { Destination, ProxyConfiguration, Protocol } from '@sap-cloud-sdk/connectivity';
+import { destination as e2eDestination } from "../test-util";
+import { testEntityApi } from "../test-utils/test-entity-operations";
+import { ErrorWithCause } from "@sap-cloud-sdk/util";
+import { AxiosError } from "axios";
+const { proxyAuth, proxyHost, proxyPort } = require('./proxy-server-config')
+
+
+const destination: Destination = {
+  url: e2eDestination?.url,
+  proxyType: 'OnPremise',
+  // NoAuthentication is not possible with OnPremise proxy type.
+  authentication: 'BasicAuthentication',
+  username: '',
+  password: '',
+  proxyConfiguration: {
+    host: proxyHost,
+    port: proxyPort,
+    protocol: Protocol.HTTP,
```

Comment:
I think the proxy used is not a web proxy but a socks proxy which is a different protocol.
We would need to include a [proxy agent](https://www.npmjs.com/package/socks-proxy-agent) in the request and include this in the [node-mailer congfig](https://nodemailer.com/smtp/proxies/#2-using-socks-proxy)

Reply:
Added a section about proxy protocols for different proxy types.